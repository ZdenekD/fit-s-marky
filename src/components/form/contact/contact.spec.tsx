import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../../__test__/utils/helpers';
import Contact from '.';

const handleSubmit = jest.fn();
const useForm = () => ({
    register: jest.fn(),
    errors: [],
    handleSubmit,
});

jest.mock('react-hook-form', () => ({useForm}));

describe('Contact form', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Contact />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-contact');

        expect(component.exists()).toBe(true);
    });

    it('renders loader if state is `disabled`', () => {
        React.useState = jest.fn(() => [true, jest.fn()]);

        const container = shallow(<Contact />);
        const component = findComponent(container, 'component-loader');

        expect(component.exists()).toBe(true);
    });

    it('run handleSubmit on form submit', () => {
        const component = findComponent(wrapper, 'component-contact');

        component.find('Button').simulate('submit');

        expect(handleSubmit).toHaveBeenCalledTimes(1);
    });
});
