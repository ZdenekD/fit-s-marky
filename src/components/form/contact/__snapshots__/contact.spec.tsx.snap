// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact form match snapshot 1`] = `
<form class="form form"
      data-test="component-form"
>
  <div class="control ">
    <label for="_5innv1"
           class="label "
    >
      Jméno
      <sup class="required">
        *
      </sup>
    </label>
    <div class="wrapper">
      <input type="text"
             id="_5innv1"
             name="name"
             class="input  "
             required
             maxlength="30"
             data-test="component-input"
      >
      <span class="maxlength ">
        0 / 30
      </span>
    </div>
  </div>
  <div class="control ">
    <label for="_5innv2"
           class="label "
    >
      E-mail
      <sup class="required">
        *
      </sup>
    </label>
    <div class="wrapper">
      <input type="email"
             id="_5innv2"
             name="email"
             class="input  "
             required
             maxlength="90"
             data-test="component-input"
      >
      <span class="maxlength ">
        0 / 90
      </span>
    </div>
  </div>
  <div class="control ">
    <label for="_5innv3"
           class="label "
    >
      Zpráva
      <sup class="required">
        *
      </sup>
    </label>
    <div class="wrapper">
      <textarea id="_5innv3"
                name="message"
                class="textarea   textareaMaxlength"
                required
                maxlength="255"
                data-test="component-textarea"
      >
      </textarea>
      <span class="maxlength ">
        0 / 255
      </span>
    </div>
  </div>
  <div class="control hidden">
    <label for="_5innv4"
           class="label "
    >
      Med
    </label>
    <div class="wrapper">
      <input type="text"
             id="_5innv4"
             name="med"
             class="input  "
             autocomplete="off"
             data-test="component-input"
      >
    </div>
  </div>
  <button type="submit"
          class="button info submit"
          data-test="component-button"
  >
    Odeslat
  </button>
</form>
`;
